src/meltano/cli/config.py

```diff
@@ -372,14 +373,25 @@ def set_(
     value: t.Any,
     store: str,
     interactive: bool,
+    from_file,
 ):
     """Set the configurations' setting `<name>` to `<value>`."""
     if len(setting_name) == 1:
         setting_name = tuple(setting_name[0].split("."))
 
     interaction = InteractiveConfig(ctx=ctx, store=store, extras=False)
+
     if interactive:
         interaction.configure_all()
+    elif from_file:
+        setting_name += (value,)
+        value = from_file.read()
+
+        interaction.set_value(
+            setting_name=setting_name,
+            value=value,
+            store=store,
+        )
     else:
         interaction.set_value(setting_name=setting_name, value=value, store=store)
```

Comment:
This can be simplified if we exit explicitly after configuring interactively:

```suggestion
    if interactive:
        interaction.configure_all()
        ctx.exit()

    if from_file:
        setting_name += (value,)
        value = from_file.read()

    interaction.set_value(setting_name=setting_name, value=value, store=store)
```